iphone/cli/commands/_build.js

```diff
@@ -2015,6 +2015,10 @@ iOSBuilder.prototype.initialize = function initialize() {
 	// TIMOB-17892
 	this.currentBuildManifest.useJSCore = this.useJSCore = !this.debugHost && !this.profilerHost && this.cli.tiapp.ios && this.cli.tiapp.ios['use-jscore-framework'];
 
+	if (this.cli.tiapp.ios && this.cli.tiapp.ios['run-on-main-thread'] !== undefined) {
```

Comment:
If you're setting anything on `this.currentBuildManifest`, you must not do it conditionally. Change this whole block to the following:

```
this.currentBuildManifest.runOnMainThread = this.runOnMainThread = this.tiapp.ios && this.tiapp.ios['run-on-main-thread'];
```